### DISCUSSION TITLE
Can we use socket socket.io-client as standalone on browser without having socket.io in server side

### DISCUSSION THREAD

**Discussion by vicky-allen** [socketio/socket.io#4886] [Q&A]
Hi Team,
We are implementing chat application where our backend is using gorilla golang
In UI can we use socket.io-client as standalone or do we need socket.io on server side as well
I am able to create a connection on UI but not able to send message 
Events like **io!.on('connection',)** not working
socket.emit('request', /* … */); // not working
socket.on('reply', (event) => { 
          console.log("socket reply", event)
         }); 
Above events are not working
What i suspect is socket-io is event based and we need socket io in backend also to receive these events in client

`import { io } from "socket.io-client";`
` 
       this.socket = io('url', {
        path: '/',
        transports: ["websocket"],
        autoConnect: true,
       // reconnectionDelayMax: 10000,
        query: {

        }
      })
     
      io!.on('connection', socket => {
        this.setupKeepAlive()
        // socket.emit('request', /* … */); // emit an event to the socket
        // io.emit('broadcast', /* … */); // emit an event to all connected sockets
        const testObject = {
          "request": {
            "type": "TEST",
            "sub_type": "TEST1",
            "data": {}
          }
        };
        this.socket!.emit('request',JSON.stringify(testObject));
        socket.on('reply', (event) => { 
          console.log("socket reply", event)
         }); // listen to the event
      });
`

**Comment by vicky-allen** [comment]
https://stackoverflow.com/questions/19506735/client-side-socket-io-without-a-node-js-server this stackoverflow question suggests that server should also have used shocket io protocol my question how can we test that server is using socket io protocol

Also can we use https://github.com/socketio/engine.io-client/ as stand alone

**Comment by darrachequesne** [comment]
Hi! A Socket.IO server is indeed needed. Several implementations are available:

| Language | Source |
|----------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------|
| JavaScript (Node.js) | https://github.com/socketio/socket.io |
| JavaScript (Deno)    | https://github.com/socketio/socket.io-deno |
| Java                 | https://github.com/mrniko/netty-socketio                                                                                                                |
| Java                 | https://github.com/trinopoty/socket.io-server-java                                                                                                      |
| Python               | https://github.com/miguelgrinberg/python-socketio                                                                                                       |
| Golang               | https://github.com/googollee/go-socket.io                                                                                                               |
| Rust                 | https://github.com/Totodore/socketioxide                                                                                                                |

Reference: https://socket.io/docs/v4/#server-implementations

The Socket.IO protocol is described there: https://socket.io/docs/v4/socket-io-protocol/
